### ISSUE TITLE
Request for the removal of content from Plex

### ISSUE THREAD

**Issue by perrottg** [petio-team/petio#687]
**Is your feature request related to a problem? Please describe.**
There is a lot of movies in the Plex library that we will never watch again but there is no easy way to do this as it needs agreement between myself and my partner.  God forbid if I remove a movie that I didn't like that my partner wold like to watch again.

I don't have delete content enabled in Plex as I don't want others in the family deleting content without my approval.  We don't usually switch the user in the Plex client so nearly all content is watched under the one user account. We also use the Infuse client in preference to the Plex client and that doesn't support Plex user switching anyway.

**Describe the solution you'd like**
1. User browses for the movie that they would like removed.
1. User selects option to request removal of the movie or TV series.
2. Removal request appears in request list for other moderators.
3. Other moderator approves or rejects the request with the ability to provide a comment. *
4. Petio automatically removes the movie/tv series (if request approved).
5. Use who requested the removal is notified of the outcome (approved/rejected).

* Petio permission scheme will define who can approve removal requests.

**Describe alternatives you've considered**
At the moment we are using the Petio report an issue feature but the above  suggestion would provide a better experience.

**Additional context**

**Comment by angrycuban13** [comment]
Petio does not do any file management and in my opinion that seems outside the scope of this project. However, as I'm not heavily involved with the code base I will let one of the devs know.

If Plex supports media removal via API, that might be able to be accomplished that way but Petio will **never** touch your files in the way you'd expect in Step 5.

The other alternative would be to send the removal request via API to Sonarr/Radarr since these are actual file management applications but this could also limit the scope of this feature.

**Comment by perrottg** [comment]
I agree with the comment regarding file management.  The expectations is that Petio would call into the Plex API (assuming the delete function is exposed) to remove the content.

Using Sonarr/Radarr as an alternative is a great fallback idea. Maybe even better as it wouldn't be reliant on enabling the delete function is Plex, which the Plex API function may be dependant on.

**Comment by marchershey** [comment]
Petio wouldn't need to do the file manipulation. 

As long as you are utilizing Sonarr & Radarr, Petio could easily make a request to delete the media through Sonarr's/Radarr's `delete()` API.

**Comment by marchershey** [comment]
This could easily be accomplished if Petio would make a `delete()` API call to Radarr/Sonarr.

[Radarr API](https://radarr.video/docs/api/#/Movie/deleteMovie) & [Sonarr API](https://github.com/Sonarr/Sonarr/wiki/Series#deleteid)
https://radarr.video/docs/api/#/Movie/deleteMovie

**Comment by marchershey** [comment]
This could be easily accomplished. [Radarr API](https://radarr.video/docs/api/#/Movie/deleteMovie) & [Sonarr API](https://github.com/Sonarr/Sonarr/wiki/Series#deleteid)

**Comment by AshDyson** [comment]
Yes, this is an interesting one, everything we do with Petio is non-destructive currently. It's a bold move to start allowing actions to trigger media deletion so it's something we'll have to carefully consider / implement. But yes either way we would use Sonarr / Radarr to manage the deletion, we treat the Plex connection as ready only (with some minor exceptions).
